server/cpho/jinja2/indicator_data/manage_indicator_data.jinja2

```diff
@@ -200,7 +200,14 @@
           </thead>
           <tbody id="{{ dimension_type }}-table-body"
                  class="text-center age-group-form-list">
-            {% for form in age_group_formset %}{{ age_group_form(form) }}{% endfor %}
+            {% for item in age_group_sorted %}
+              {% for form_instance in age_group_formset %}
+                {% if form_instance.instance.id == item.id %}
+                  {% set form = form_instance %}
+                  {{ age_group_form(form) }}
```

Comment:
suuuper nitpick, is it worth creating a variable? Why not just `age_group_form(form_instance)`

Reply:
Have you tested this with new records that are invalid? E.g. POST a form with new invalid records. They wouldn't be in the querset, are they still rendered?

Reply:
Yeah this is a bug, maybe we can follow the loop with this one,
```jinja2
{% for form_instance in age_group_formset %}
  {% if not form_instance.instance.id %}
    {{age_group_form(form_instance)}}
  {% endif %}
{% endfor %}
```

Reply:
Good catch
i pushed up the changes
Thanks
:)